src/libraries/System.Security.Cryptography/tests/X509Certificates/PfxIterationCountTests.cs

```diff
@@ -0,0 +1,227 @@
+
+        internal static List<PfxInfo> GetCertificates()
+        {
+            List<PfxInfo> certificates = new List<PfxInfo>();
+            certificates.Add(new PfxInfo(
+                nameof(TestData.Pkcs12NoPassword2048RoundsHex), null, 2048 * 3, true, TestData.Pkcs12NoPassword2048RoundsHex.HexToByteArray()));
+            certificates.Add(new PfxInfo(
+                nameof(TestData.Pkcs12OpenSslOneCertDefaultEmptyPassword), "", 2048 * 3, true, TestData.Pkcs12OpenSslOneCertDefaultEmptyPassword.HexToByteArray()));
+            certificates.Add(new PfxInfo(
+                nameof(TestData.Pkcs12OpenSslOneCertDefaultNoMac), null, 2048, true, TestData.Pkcs12OpenSslOneCertDefaultNoMac.HexToByteArray()));
+            certificates.Add(new PfxInfo(
+                nameof(TestData.Pkcs12NoPasswordRandomCounts), null, 938, true, TestData.Pkcs12NoPasswordRandomCounts));
+            certificates.Add(new PfxInfo(
+                nameof(TestData.Pkcs12WindowsDotnetExportEmptyPassword), "", 6000, false, TestData.Pkcs12WindowsDotnetExportEmptyPassword.HexToByteArray()));
+            certificates.Add(new PfxInfo(
+                nameof(TestData.Pkcs12MacosKeychainCreated), null, 4097, false, TestData.Pkcs12MacosKeychainCreated.HexToByteArray()));
+            certificates.Add(new PfxInfo(
+                nameof(TestData.Pkcs12BuilderSaltWithMacNullPassword), null, 120000, true, TestData.Pkcs12BuilderSaltWithMacNullPassword.HexToByteArray()));
+            certificates.Add(new PfxInfo(
+                nameof(TestData.Pkcs12Builder3DESCBCWithNullPassword), null, 30000, false, TestData.Pkcs12Builder3DESCBCWithNullPassword.HexToByteArray()));
+            certificates.Add(new PfxInfo(
+                nameof(TestData.Pkcs12Builder3DESCBCWithEmptyPassword), "", 30000, false, TestData.Pkcs12Builder3DESCBCWithEmptyPassword.HexToByteArray()));
+            certificates.Add(new PfxInfo(
+                nameof(TestData.Pkcs12WindowsWithCertPrivacyPasswordIsOne), "1", 4000, false, TestData.Pkcs12WindowsWithCertPrivacyPasswordIsOne.HexToByteArray()));
+            certificates.Add(new PfxInfo(
+                nameof(TestData.Pkcs12WindowsWithoutCertPrivacyPasswordIsOne), "1", 4000, false, TestData.Pkcs12WindowsWithoutCertPrivacyPasswordIsOne.HexToByteArray()));
+            certificates.Add(new PfxInfo(
+                nameof(TestData.Pkcs12NoPassword600KPlusOneRoundsHex), null, 600_001 * 3, true, TestData.Pkcs12NoPassword600KPlusOneRoundsHex.HexToByteArray()));
+
+            return certificates;
+        }
+
+        public static IEnumerable<object[]> GetCertsWith_IterationCountNotExceedingDefaultLimit_AndNullOrEmptyPassword_MemberData()
+        {
+            foreach (PfxInfo p in s_Certificates.Where(
+                c => c.IterationCount <= DefaultIterationLimit &&
+                string.IsNullOrEmpty(c.Password)))
+            {
+                yield return new object[] { p.Name, p.UsesPbes2, p.Blob, p.IterationCount };
+            }
```

Comment:
For each of these foreach .Where, it really seems like the LINQ Where isn't buying much over just writing an if inside the foreach.  LINQ Where is generally only useful when a) followed by something else, like Select/GroupBy/OrderBy/etc or b) when passed into a method as an IEnumerable(`<T>`) parameter.

```suggestion
            foreach (PfxInfo p in s_Certificates)
            {
                if (p.IterationCount <= DefaultIterationLimit && string.IsNullOrEmpty(p.Password))
                {
                    yield return new object[] { p.Name, p.UsesPbes2, p.Blob, p.IterationCount };
                }
            }
```

Unless Maoni is giving you a dollar for every object the GC has to collect... in which case, sure :smile: